flocker/node/agents/test/test_cinder.py

```diff
@@ -61,3 +67,65 @@ def test_verify_false_ca_path(self):
             'verify_ca_path': '/a/path'
         }
         self.assertEqual(_openstack_verify_from_config(**config), False)
+
+
+class FakeCinderClient(object):
+    """"
+    Fake implementation of the cinder volume manager to use in the Cinder
+    Destroy tests.
+    Right now, we don't need a full mock of the class, we just need the
```

Comment:
in general, we're probably going to try not to use mocks anywhere in Flocker's tests (at least, that's my inclination).  I find other strategies almost always produce better tests.

I point this out because it's easy for people to mix up the different kinds of test fake strategies so wherever possible we should be clear on the terms we use and make sure we're all on the same page about what they mean.

In this case, I'd say "we don't need a full fake" (or a "complete fake", reads slightly more easily).

It may be that the deletion timeout logic never needs something more complete than this.  Perhaps this isn't just "FakeCinderClient", perhaps this is "SlowDeletionCinderClient" ("Slow" doesn't quite capture it since it actually never deletes things you ask it to... "SilentDeletionFailureCinderClient" is maybe too verbose a class name though).

So maybe it's worth saying (and committing to) this is only a fake suitable for those particular deletion tests and if we need some other kind of behavior we'll build another fake for it.